internal/components/inspektorgadget/gadgetmanager.go

```diff
@@ -115,20 +118,43 @@ func (g *manager) outputDataOperator(cb func(data []byte)) operators.DataOperato
 	return simple.New("outputDataOperator",
 		simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
 			for _, d := range gadgetCtx.GetDataSources() {
-				jsonFormatter, _ := igjson.New(d,
-					// TODO: Maybe we should use specific fields or remove unwanted fields
-					igjson.WithShowAll(true),
-				)
-
 				// skip data sources that have the annotation "cli.default-output-mode"
 				// set to "none"
 				if m, ok := d.Annotations()["cli.default-output-mode"]; ok && m == "none" {
 					continue
 				}
 
-				err := d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				// handle adding a raw string field for certain content types
+				restField := d.Annotations()["ebpf.rest.name"]
+				var restAcc datasource.FieldAccessor
+				var restStrAcc datasource.FieldAccessor
+				var err error
+				if restField != "" {
+					restAcc = d.GetField(restField)
+					ct, ok := restAcc.Annotations()["content-type"]
+					if ok && ct == "application/x-raw-packet" {
+						restStrAcc, err = d.AddField(restField+"_string", api.Kind_String)
+						if err != nil {
+							return fmt.Errorf("adding raw string field accessor: %w", err)
+						}
+					}
+				}
+
+				jsonFormatter, _ := igjson.New(d,
```

Comment:
nit: shall we check the error when creating jsonFormatter (though it is not changed this time)?

Reply:
Done, nice catch! 

> shall we check the error when creating jsonFormatter (though it is not changed this time)?

Not sure why linter didn't catch it before but good to have it!